### PR TITLE
github signup flow with github data fetched

### PR DESCRIPTION
**Pull request by asood123**
Still need to connect the collect info on the frontend (like repo name, contributors, mission statement, etc) and create a collective and users on the backend. Will save that for another PR.

### CODE REVIEW

File: frontend/src/components/on_boarding/OnBoardingStepPickCoreContributors.js [hunk 1]
```diff
@@ -25,17 +28,19 @@ export default class OnBoardingStepPickCoreContributors extends React.Component
...
             chosenContributors.splice(chosenIndex, 1);
             availableContributors.push(contributor);
+            appendGithubForm({contributors: chosenContributors.map((c) => c.name)});
             this.forceUpdate();
           }}
         />
-        <div style={{margin: '0 auto', marginTop: '40px', width: '300px', textAlign: 'center'}}>
-          <div className={`OnBoardingButton ${chosenContributors.length >= MIN_CONTRIBUTORS ? '' : 'disabled'}`} onClick={chosenContributors.length >= MIN_CONTRIBUTORS ? onNextStep(chosenContributors) : null }>continue</div>
+        <div style={buttonContainerStyle}>
+          <div className={`OnBoardingButton ${contributors.length >= MIN_CONTRIBUTORS_FOR_ONBOARDING-1 ? '' : 'disabled'}`} onClick={contributors.length >= MIN_CONTRIBUTORS_FOR_ONBOARDING-1 ? () => onNextStep(contributors) : null }>continue</div>
```
Comment (carlosascari) [inline frontend/src/components/on_boarding/OnBoardingStepPickCoreContributors.js]: This must be from previous commits it should not be minus one.

File: frontend/src/containers/OnBoarding.js [hunk 2]
```diff
@@ -42,58 +54,55 @@ export class OnBoarding extends Component {
...
+    const attr = githubForm.attributes;
     const payload = {
       group: {
-        name: selectedRepo,
-        slug: selectedRepo,
-        mission: missionDescription,
-        expensePolicy: expenseDescription,
-        logo
+        name: attr.repository,
+        slug: attr.selectedRepo,
```
Comment (carlosascari) [inline frontend/src/containers/OnBoarding.js]: selectedRepo is a param name the attribute property name is `repository` now

should be
```
slug: attr.repository
```

File: frontend/src/actions/github/fetch_repos.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { getThirdParty } from '../../lib/api';
+import * as constants from '../../constants/github';
+
+
+/**
+ * Fetch a user's repo
+ */
+
+export default (username) => {
```
Comment (sedubois) [inline frontend/src/actions/github/fetch_repos.js]: These action files are very repetitive/copy-pasted, have a generic structure of request/success/failure. Might benefit from e.g a prototype or template.

File: frontend/src/components/on_boarding/OnBoardingStepConnectGithub.js [hunk 4]
```diff
@@ -8,15 +8,16 @@ export default class OnBoardingStepConnectGithub extends React.Component {
...
   render()
   {
-    const { onNextStep } = this.props;
     return (
       <div className="OnBoardingStepConnectGithub">
-          <OnBoardingStepHeading step="1/4" title="Connect your Github account."/>
+          <OnBoardingStepHeading step="1/4" title="Connect your GitHub account"/>
           <div style={{margin: '0 auto', marginTop: '40px', width: '300px', textAlign: 'center'}}>
-            <a href="#" className="OnBoardingStepButton mb3 -btn -btn-big -bg-green -ttu -ff-sec -fw-bold color-white m-auto" onClick={onNextStep}>connect account</a>
+            <a href='/api/connected-accounts/github'>
```
Comment (sedubois) [inline frontend/src/components/on_boarding/OnBoardingStepConnectGithub.js]: `ConnectAccountButton.js` should be deleted or injected here.

File: frontend/src/components/on_boarding/OnBoardingStepPickRepository.js [hunk 5]
```diff
@@ -1,30 +1,37 @@
 import React from 'react';
 
+import { MIN_STARS_FOR_ONBOARDING } from '../../constants/github';
+
 import OnBoardingStepHeading from './OnBoardingStepHeading';
 import RepositoryPicker from './RepositoryPicker';
 
 export default class OnBoardingStepPickRepository extends React.Component {
```
Comment (sedubois) [inline frontend/src/components/on_boarding/OnBoardingStepPickRepository.js]: This component doesn't have any internal state -> recommend functional notation

File: frontend/src/containers/OnBoarding.js [hunk 6]
```diff
@@ -11,34 +11,131 @@ import OnBoardingStepPickCoreContributors from '../components/on_boarding/OnBoar
...
         <Notification />
-        {step !== 5 && <OnBoardingHeader active={Boolean(step)} />}
+        {step !== 5 && <OnBoardingHeader active={Boolean(step)} username={githubForm.attributes.username} />}
         {step === 0 && <OnBoardingHero onClickStart={() => this.setState({step: 1})} />}
-        {step === 1 && <OnBoardingStepConnectGithub onNextStep={() => this.setState({step: 2})} />}
-        {step === 2 && <OnBoardingStepPickRepository onNextStep={() => this.setState({step: 3})} />}
-        {step === 3 && <OnBoardingStepPickCoreContributors onNextStep={() => this.setState({step: 4})} />}
-        {step === 4 && <OnBoardingStepCreate onCreate={() => this.setState({step: 5})}/>}
-        {step === 5 && <OnBoardingStepThankYou onContinue={() => this.setState({step: 0})}/>}
+        {step === 1 && <OnBoardingStepConnectGithub />}
```
Comment (sedubois) [inline frontend/src/containers/OnBoarding.js]: Use a `switch` for all these steps?

Comment (asood123) [inline frontend/src/containers/OnBoarding.js]: Tried doing that but it requires the entire statement to be in quotes and makes it hard to read. If you know of another way of doing this in JSX, let me know.

File: frontend/src/components/on_boarding/OnBoardingStepConnectGithub.js (hunk 4 above)
Comment (asood123) [inline frontend/src/components/on_boarding/OnBoardingStepConnectGithub.js]: Made this stateless.

File: frontend/src/components/on_boarding/OnBoardingStepPickRepository.js (hunk 5 above)
Comment (asood123) [inline frontend/src/components/on_boarding/OnBoardingStepPickRepository.js]: done

File: frontend/src/actions/github/fetch_repos.js (hunk 3 above)
Comment (asood123) [inline frontend/src/actions/github/fetch_repos.js]: I'm sure there is a way to do this - going to leave this for another time :)

File: test/e2e/connect_account_page.js [hunk 7]
```diff
@@ -19,9 +19,6 @@ module.exports = {
     // const githubScope = encodeURIComponent('user:email');
 
     client
-      .click('.connectAccountBtn')
-      .waitForElementVisible('body', 4000)
-      .assert.urlContains('https://github.com/login?return_to=%2Flogin%2Foauth%2Fauthorize')
```
Comment (sedubois) [inline test/e2e/connect_account_page.js]: Why is this removed?

Comment (asood123) [inline test/e2e/connect_account_page.js]: Oh you are right. that last line should have stayed. I'll add it back.
